# ESLint Plugin: eslint-plugin-no-explicit-undefined

Disallow the use of optional operator (`?`)  as an explicit type in TypeScript and instead use `undefined` instead.

## ğŸ“š Rule Examples

This rule disallows the use of optional operator (`?`)as an explicit type in TypeScript. Instead, developers should use `undefined`.

Examples of **incorrect** code for this rule âŒ

```ts
interface Example { prop?: string; }
type Example = { prop?: string; }
```

Examples of **correct** code for this rule âœ…

```ts
interface Example { prop: string | undefined; }
type Example = { prop: string | undefined; }
```

## ğŸš€ Installation

Install the package using `npm`:

```bash
npm install eslint-plugin-no-implicit-undefined --save-dev
```

## ğŸ“ Configuration

To configure the rule, add it to your ESLint configuration file (`.eslintrc.json`, `.eslintrc.js`, etc.):

```json
{
  "extends": ["custom"],
  "plugins": ["no-optional-type-declaration"],
  "rules": {
    "no-optional-type-declaration/no-implicit-undefined": "error"
  }
}
```

## ğŸ“ƒ License

MIT
